addons/sale/models/product_template.py

```diff
@@ -93,3 +145,325 @@ def get_import_templates(self):
+
+        if not product_id and not combination:
+            combination = product_template._get_first_possible_combination(reference_product)
+
+        if product_id and not combination:
+            product = product_template.env['product.product'].browse(product_id)
+        else:
+            # no_variant attributes can still be on a variant if their type has
+            # been changed after the creation of the variant
+            product = product_template._get_variants_for_combination(combination, exclude_no_variant=False)[:1]
+            if not product:
+                product = product_template._get_variants_for_combination(combination)[:1]
+
+        if product:
+            # We need to add the price_extra for the attributes that are not
+            # in the variant, typically those of type no_variant, but it is
+            # possible that a no_variant attribute is still in a variant if
+            # the type of the attribute has been changed after creation.
+            product = product.with_context(
+                no_variant_attributes_price_extra=[ptav.price_extra or 0.0
+                    for ptav in combination.filtered(
+                        lambda ptav: ptav not in product.product_template_attribute_value_ids
+                    )
+                ]
+            )
+            list_price = product.price_compute('list_price')[product.id]
+            price = product.price if pricelist else list_price
+        else:
+            product_template = product_template.with_context(current_attributes_price_extra=[v.price_extra or 0.0 for v in combination])
+            list_price = product_template.price_compute('list_price')[product_template.id]
+            price = product_template.price if pricelist else list_price
+
+        display_name = product_template.name
+
+        filtered_combination = combination._filtered_no_variant()
+        if filtered_combination:
+            display_name = '%s (%s)' % (display_name, ', '.join(filtered_combination.mapped('name')))
+
+        if pricelist and pricelist.currency_id != product_template.currency_id:
+            list_price = product_template.currency_id._convert(
+                list_price, pricelist.currency_id, product_template.company_id,
+                fields.Date.today()
+            )
+
+        price_without_discount = list_price if pricelist and pricelist.discount_policy == 'without_discount' else price
+        has_discounted_price = (pricelist or product_template).currency_id.compare_amounts(price_without_discount, price) == 1
+
+        return {
+            'product_id': product.id,
+            'product_template_id': product_template.id,
+            'display_name': display_name,
+            'price': price,
+            'list_price': list_price,
+            'has_discounted_price': has_discounted_price,
+        }
+
+    @api.multi
+    def _get_variants_for_combination(self, combination, active_test=True, exclude_no_variant=True):
```

Comment:
@seb-odoo You use [:1] in every call to this method. Wouldn't be nice to create a "_get_variant_for_combination" that does it for clarity?

Reply:
I actually created the single version originally and then I realized the same query could be used to get multiple... But this is a good idea to make a single version so I can even add a limit 1 on the query to optimize it.